core/bentley/src/OneAtATimeAction.ts

```diff
@@ -83,8 +83,7 @@ export class OneAtATimeAction<T> {
     }
 
     try {
-      await promise;
-      return promise;
+      return await promise;

```

Comment:
@bbastings think this is relevant to the crash while snapping we observed ~a month ago?